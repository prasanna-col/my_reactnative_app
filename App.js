/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, { useEffect, useState } from 'react';
import type { Node } from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  Image,
  Button,
  TextInput,
  useColorScheme,
  View,
} from 'react-native';
import InsetShadow from 'react-native-inset-shadow'
// import FlashMessage from "react-native-flash-message";
import { showMessage, hideMessage } from "react-native-flash-message";

import {
  Colors,

} from 'react-native/Libraries/NewAppScreen';
import { CirclesLoader, PulseLoader, TextLoader, DotsLoader } from 'react-native-indicator';
import RNQRGenerator from 'rn-qr-generator';


const App: () => Node = () => {
  const isDarkMode = useColorScheme() === 'dark';
  const [refreshOrders, setRefreshOrders] = useState(true);

  const [ImageURI,setImageURI] = useState("");

  useEffect(() => {
    setTimeout(() => {
      setRefreshOrders(!refreshOrders)
    }, 5000)
  }, [])

  const get_qrcode = () => {
    RNQRGenerator.generate({
      value: 'https://github.com/gevgasparyan/rn-qr-generator',
      height: 100,
      width: 100,
    })
      .then(response => {
        const { uri, width, height, base64 } = response;
        // this.setState({ imageUri: uri });
        setImageURI(uri)
        console.log("imageUri", ImageURI)
      })
      .catch(error => console.log('Cannot create QR code', error));

    // Detect QR code in image
    // RNQRGenerator.detect({
    //   uri: PATH_TO_IMAGE
    // })
    //   .then(response => {
    //     const { values } = response; // Array of detected QR code values. Empty if nothing found.
    //   })
    //   .catch(error => console.log('Cannot detect QR code in image', error));
  }

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  return (
    <View style={backgroundStyle, [{ padding: 20, backgroundColor: "#fff", height: "100%", width: "100%" }]}>

      {
        refreshOrders ?
          <View style={{ height: "100%", width: "100%" }}>
            <CirclesLoader />
            <TextLoader text="Loading" />
          </View>
          :
          <View style={{ height: "100%", width: "100%" }}>
            <View>
            <TextInput style={{
              paddingLeft: 20,
              backgroundColor: 'transparent',
              borderWidth: 1.5,
              borderRadius: 5,
              borderColor: "red",
              shadowColor: 'red',
              shadowOffset: { width: 0, height: 1 },
              shadowOpacity: 0.2,
              elevation: 1
            }}>

            </TextInput>
            <TextInput style={{
              top: 50,
              paddingLeft: 20,

              backgroundColor: '#fff',
              borderColor: "orange",
              borderWidth: 2,
              borderRadius: 5,

              elevation: 10,
              shadowColor: 'orange',
              shadowOpacity: 4,
              shadowOffset: {
                width: 2,
                height: 2
              },
              shadowRadius: 10, // <- Radius of the shadow
            }}>

            </TextInput>
            </View>
            <View>
              <Button
                onPress={() => {
                  /* HERE WE GONE SHOW OUR FIRST MESSAGE */
                  // showMessage({
                  //   message: "Simple message",
                  //   type: "info",
                  // });

                  get_qrcode();
                }}
                title="Request Details"
                color="#841584"
              />
            </View>

            <View>
              <Image source={{uri:ImageURI}} style={{height:100, width:100}} />
            </View>

          </View>
      }

    </View>
  );
};

const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  sectionTitle: {
    fontSize: 24,
    fontWeight: '600',
  },
  sectionDescription: {
    marginTop: 8,
    fontSize: 18,
    fontWeight: '400',
  },
  highlight: {
    fontWeight: '700',
  },
});

export default App;
