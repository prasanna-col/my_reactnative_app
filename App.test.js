import React from 'react'
import 'react-native'
import renderer from "react-test-renderer"
import App from './App'

test('Check the landing page', () => {
    const tree = renderer.create(<App/>).toJSON()
    expect(tree).toMatchSnapshot()
});