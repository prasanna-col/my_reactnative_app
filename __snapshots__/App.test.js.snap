// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the landing page 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "height": "100%",
        "padding": 20,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <ARTSurfaceView
      style={
        Array [
          undefined,
          Object {
            "height": 48,
            "width": 48,
          },
        ]
      }
    >
      <ARTShape
        d={
          Array [
            0,
            0,
            -4,
            4,
            0,
            0,
            4,
            -1.5707963267948966,
            1.5707963267948966,
            1,
            4,
            0,
            0,
            4,
            1.5707963267948966,
            -1.5707963267948966,
            1,
            1,
          ]
        }
        fill={
          Array [
            0,
            0.11764705882352941,
            0.5647058823529412,
            1,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            44,
            24,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            0,
            -4,
            4,
            0,
            0,
            4,
            -1.5707963267948966,
            1.5707963267948966,
            1,
            4,
            0,
            0,
            4,
            1.5707963267948966,
            -1.5707963267948966,
            1,
            1,
          ]
        }
        fill={
          Array [
            0,
            0.11764705882352941,
            0.5647058823529412,
            1,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            38,
            38,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            0,
            -4,
            4,
            0,
            0,
            4,
            -1.5707963267948966,
            1.5707963267948966,
            1,
            4,
            0,
            0,
            4,
            1.5707963267948966,
            -1.5707963267948966,
            1,
            1,
          ]
        }
        fill={
          Array [
            0,
            0.11764705882352941,
            0.5647058823529412,
            1,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            24,
            44,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            0,
            -4,
            4,
            0,
            0,
            4,
            -1.5707963267948966,
            1.5707963267948966,
            1,
            4,
            0,
            0,
            4,
            1.5707963267948966,
            -1.5707963267948966,
            1,
            1,
          ]
        }
        fill={
          Array [
            0,
            0.11764705882352941,
            0.5647058823529412,
            1,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            10,
            38,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            0,
            -4,
            4,
            0,
            0,
            4,
            -1.5707963267948966,
            1.5707963267948966,
            1,
            4,
            0,
            0,
            4,
            1.5707963267948966,
            -1.5707963267948966,
            1,
            1,
          ]
        }
        fill={
          Array [
            0,
            0.11764705882352941,
            0.5647058823529412,
            1,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            4,
            24,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            0,
            -4,
            4,
            0,
            0,
            4,
            -1.5707963267948966,
            1.5707963267948966,
            1,
            4,
            0,
            0,
            4,
            1.5707963267948966,
            -1.5707963267948966,
            1,
            1,
          ]
        }
        fill={
          Array [
            0,
            0.11764705882352941,
            0.5647058823529412,
            1,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            10,
            10,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            0,
            -4,
            4,
            0,
            0,
            4,
            -1.5707963267948966,
            1.5707963267948966,
            1,
            4,
            0,
            0,
            4,
            1.5707963267948966,
            -1.5707963267948966,
            1,
            1,
          ]
        }
        fill={
          Array [
            0,
            0.11764705882352941,
            0.5647058823529412,
            1,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            24,
            4,
          ]
        }
      />
      <ARTShape
        d={
          Array [
            0,
            0,
            -4,
            4,
            0,
            0,
            4,
            -1.5707963267948966,
            1.5707963267948966,
            1,
            4,
            0,
            0,
            4,
            1.5707963267948966,
            -1.5707963267948966,
            1,
            1,
          ]
        }
        fill={
          Array [
            0,
            0.11764705882352941,
            0.5647058823529412,
            1,
            1,
          ]
        }
        opacity={1}
        stroke={null}
        strokeCap={1}
        strokeDash={null}
        strokeJoin={1}
        strokeWidth={1}
        transform={
          Array [
            1,
            0,
            0,
            1,
            38,
            10,
          ]
        }
      />
    </ARTSurfaceView>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text>
        Loading
      </Text>
      <Text
        style={
          Array [
            Object {
              "opacity": 0,
            },
            undefined,
          ]
        }
      >
        .
      </Text>
      <Text
        style={
          Array [
            Object {
              "opacity": 0,
            },
            undefined,
          ]
        }
      >
        .
      </Text>
      <Text
        style={
          Array [
            Object {
              "opacity": 0,
            },
            undefined,
          ]
        }
      >
        .
      </Text>
    </View>
  </View>
</View>
`;
